src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2821,245 +2821,176 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
-        std::vector<LedgerEntry> ledgerVect{entrySet.begin(), entrySet.end()};
-        std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(),
-                                           deadKeySet.end()};
+    auto deadKeys =
+        LedgerTestUtils::generateValidUniqueLedgerEntryKeysWithExclusions(
+            {CONFIG_SETTING}, 1);
+    for (const auto& k : deadKeys)
+    {
+        ltx.eraseWithoutLoading(k);
+        deadKeySet.emplace(k);
+    }
 
-        app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect,
-                                         deadKeyVect);
-        ltx.commit();
+    // Insert all entries into the database.
+    std::vector<LedgerEntry> ledgerVect{classicEntry, contractDataEntry,
+                                        TTLEntry};
+    std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(), deadKeySet.end()};
 
-        auto addTxn = [&](int32_t offset, std::vector<LedgerEntry> entries,
-                          std::vector<LedgerKey> deadKeys = {}) {
-            SorobanResources resources;
+    app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect, deadKeyVect);
+    ltx.commit();
 
-            for (auto& e : entries)
+    auto addTxn = [&](bool enoughQuota, std::vector<LedgerEntry> entries,
+                      std::vector<LedgerKey> deadKeys = {}) {
+        SorobanResources resources;
+
+        for (auto const& e : entries)
+        {
+            auto k = LedgerEntryKey(e);
+            if (k.type() != TTL)
             {
                 resources.readBytes += xdr::xdr_size(e);
-                auto k = LedgerEntryKey(e);
-                keysToPrefetch.emplace(k);
-                // Randomly add the key to either the read or write set.
-                if (stellar::rand_flip())
-                {
-                    resources.footprint.readOnly.emplace_back(k);
-                }
-                else
-                {
-                    resources.footprint.readWrite.emplace_back(k);
-                }
-                // If the key is not a TTL and its offset is less than 0 and
-                // it has not been marked as expected to succeed in another
-                // transaction, add it to the expected failed keys set.
-                if (k.type() != TTL && offset < 0 &&
-                    expectedSuccessKeys.find(k) == expectedSuccessKeys.end())
-                {
-                    expectedFailedKeys.emplace(k);
-                }
-                else
-                {
-                    expectedSuccessKeys.emplace(k);
-                    // If a key has sufficient in one transaction, remove it
-                    // from the expected failed keys set.
-                    expectedFailedKeys.erase(k);
-                }
             }
-            if (offset + resources.readBytes < 0)
+            keysToPrefetch.emplace(k);
+            // Randomly add the key to either the read or write set.
+            if (stellar::rand_flip())
```

Comment:
While TTL keys should be in `keysToPrefetch`, they should not be in the footprint.